examples/IMUKittiExampleGPS.cpp

```diff
@@ -0,0 +1,293 @@
+    vector<imuMeasurement_t> IMU_measurements;
+    string IMU_data_file = findExampleDataFile("KittiEquivBiasedImu.txt");
+
+    printf("-- Reading IMU measurements from file\n");
+    {
+        ifstream IMU_data(IMU_data_file.c_str());
+        getline(IMU_data, line, '\n'); // ignore the first line
+
+        double time = 0, dt = 0, acc_x = 0, acc_y = 0, acc_z = 0, gyro_x = 0, gyro_y = 0, gyro_z = 0;
+        while (!IMU_data.eof()) {
+            getline(IMU_data, line, '\n');
+            sscanf(line.c_str(), "%lf %lf %lf %lf %lf %lf %lf %lf", &time, &dt, &acc_x, &acc_y, &acc_z, &gyro_x,
+                   &gyro_y, &gyro_z);
+
+            imuMeasurement_t measurement;
+            measurement.Time = time;
+            measurement.dt = dt;
+            measurement.Accelerometer = Vector3(acc_x, acc_y, acc_z);
+            measurement.Gyroscope = Vector3(gyro_x, gyro_y, gyro_z);
+            IMU_measurements.push_back(measurement);
+        }
+    }
+
+    // Read GPS data
+    // Time,X,Y,Z
+    // 46534.478375790000428,-6.8269361350059405424,-11.868164241239471224,0.040306091310000624617
+    vector<gpsMeasurement_t> GPS_measurements;
+    string GPS_data_file = findExampleDataFile("KittiGps_converted.txt");
+
+    printf("-- Reading GPS measurements from file\n");
+    {
+        ifstream GPS_data(GPS_data_file.c_str());
+        getline(GPS_data, line, '\n'); // ignore the first line
+
+        double time = 0, gps_x = 0, gps_y = 0, gps_z = 0;
+        while (!GPS_data.eof()) {
+            getline(GPS_data, line, '\n');
+            sscanf(line.c_str(), "%lf,%lf,%lf,%lf", &time, &gps_x, &gps_y, &gps_z);
+
+            gpsMeasurement_t measurement;
+            measurement.Time = time;
+            measurement.Position = Vector3(gps_x, gps_y, gps_z);
+            GPS_measurements.push_back(measurement);
+        }
+    }
+
+    // Configure different variables
+    double tOffset = GPS_measurements[0].Time;
+    size_t firstGPSPose = 1;
+    size_t GPSskip = 10; // Skip this many GPS measurements each time
+    double g = 9.8;
+    auto w_coriolis = Vector3(); // zero vector
+
+    // Configure noise models
+    noiseModel::Diagonal::shared_ptr noiseModelGPS = noiseModel::Diagonal::Precisions((Vector(6) << Vector3::Constant(0), Vector3::Constant(1.0/0.07)).finished());
+
+    // Set initial conditions for the estimated trajectory
+    auto currentPoseGlobal = Pose3(Rot3(), GPS_measurements[firstGPSPose].Position); // initial pose is the reference frame (navigation frame)
+    auto currentVelocityGlobal = Vector3(); // the vehicle is stationary at the beginning at position 0,0,0
+    auto currentBias = imuBias::ConstantBias(); // init with zero bias
+
+    noiseModel::Diagonal::shared_ptr sigma_init_x = noiseModel::Isotropic::Precisions((Vector(6) << Vector3::Constant(0), Vector3::Constant(1.0)).finished());
```

Comment:
auto

Also, (Vector(6) << Vector3::Constant(0), Vector3::Constant(1.0)).finished() is such a common case in this file and others maybe it's worth adding a small function at the top. PS It should be Diagonal ! This is actually crazy (and my fault) that you can do this: the constructor should be deleted from Isotropic.

Reply:
I have addressed the Isotropic to Diagonal part in b48bdfdc696328be3fedfe11db18b83d3341e7bc
The Vector(6) remains the same to be explicit.

Reply:
I have addressed the Isotropic to Diagonal part in b48bdfdc696328be3fedfe11db18b83d3341e7bc
The Vector(6) remains the same to be explicit.